api/src/main/java/jakarta/data/metamodel/TextExpression.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.metamodel.constraint.Like;
+import jakarta.data.metamodel.restrict.Restriction;
+
+public interface TextExpression<T> extends ComparableExpression<T,String> {
+
+    TextExpression<T> prepend(String string);
+    TextExpression<T> append(String string);
+    TextExpression<T> prepend(Expression<T,String> string);
+    TextExpression<T> append(Expression<T,String> string);
+
+    TextExpression<T> upper();
+    TextExpression<T> lower();
+
+    TextExpression<T> left(int length);
+    TextExpression<T> right(int length);
+
+    Expression<T,Integer> length();
```

Comment:
```suggestion
    NumericExpression<T,Integer> length();
```

So that you can write

        Restriction<Book> titleLT50 = _Book.title.length().lessThan(50);

instead of:

        Restriction<Book> titleLT50 = _Book.title.length().satisfies(Constraint.lessThan(50));